### DISCUSSION TITLE
Is it possible to centralize node_modules to another Drive/Disk?

### DISCUSSION THREAD

**Discussion by is-jonreeves** [pnpm/pnpm#3716] [Q&A] [answered]
I've recently been looking for a good way to centralize my `node_modules` to a Ramdisk to improve performance on my machine. I stumbled across pnpm the other day and it looks like a great tool that does a lot of what I was after, but I ran into `cross-device link not permitted` when trying to use it with a separate drive.

I'm guessing this is a OS limitation on how linking can only be done within the same filesystem, but was curious if any one knows a good work around, or another approach I could try to offload all my `node_modules` to a separate drive.

I'm using windows 10 specifically, but I'm guessing the limitation would be the same on linux or mac too.

**Comment by zkochan** [comment, marked as answer]
What you are trying to do is not possible. The store should be in the same file system as the node_modules. If they are not, pnpm will copy the files.

On some systems, where FUSE is supported, you may try to mount a node_modules directory using this experimental package: https://github.com/pnpm/pnpm/blob/main/packages/mount-modules/README.md
